Gopkg.lock

```diff
@@ -120,6 +120,12 @@
   packages = ["unix","windows"]
   revision = "fb4cac33e3196ff7f507ab9b2d2a44b0142f5b5a"
 
+[[projects]]
+  branch = "v2"
+  name = "gopkg.in/yaml.v2"
+  packages = ["."]
+  revision = "cd8b52f8269e0feb286dfeef29f8fe4d5b397e0b"
+
```

Comment:
This PR did in fact start out using templates but it got slightly unwieldy, hence the simpler marshaling code.